Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -273,40 +273,37 @@ public class ByteBuffer: Codable {
     }
 }
 
-extension ByteBuffer: AwsStream {
-    public var status: aws_stream_status {
-        return aws_stream_status(is_end_of_stream: self.currentIndex == array.count, is_valid: true)
-    }
+extension ByteBuffer: IStreamable {
 
-    public var length: UInt {
-        return UInt(array.count)
+    public func length() -> UInt64 {
```

Comment:
Nit: ooc why change to a function type? imo it'd be preferable to keep as a computed var to stay consistent with how Swift models a similar property on it's structures like Array.count, Data.count, String.count etc. 

Also nit, `count` would be a preferable name here to stay consistent with Swift naming, such as UnsafeMutableBufferPointer.count.

Reply:
Some implementations of length can throw like the file handle, so we can't have simple var.